### PR TITLE
MINOR: make sure all fiedls of o.p.k.s.a.Action are NOT null

### PR DESCRIPTION
**Pull request by chia7712**
I'm migrating Ranger's kafka plugin from deprecated Authorizer (this is already removed by 976e78e405d57943b989ac487b7f49119b0f4af4)  to new API (see https://issues.apache.org/jira/browse/RANGER-3231). The kafka plugin needs to take something from field `resourcePattern` but it does not know whether the field is nullable (or users need to add null check). I check all usages and I don't observe any null case.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/server/authorizer/Action.java [hunk 1]
```diff
@@ -36,22 +36,22 @@ public Action(AclOperation operation,
...
+        this.operation = Objects.requireNonNull(operation, "operation can't be null");
+        this.resourcePattern = Objects.requireNonNull(resourcePattern, "resourcePattern can't be null");
         this.logIfAllowed = logIfAllowed;
         this.logIfDenied = logIfDenied;
         this.resourceReferenceCount = resourceReferenceCount;
     }
 
     /**
-     * Resource on which action is being performed.
+     * Resource on which action is being performed. never null
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/server/authorizer/Action.java]: I would include the non-null constraint in the sentence, eg "Returns a non-null resource pattern on which this action is being performed".

File: clients/src/main/java/org/apache/kafka/server/authorizer/Action.java [hunk 2]
```diff
@@ -36,22 +36,22 @@ public Action(AclOperation operation,
                   int resourceReferenceCount,
                   boolean logIfAllowed,
                   boolean logIfDenied) {
-        this.operation = operation;
-        this.resourcePattern = resourcePattern;
+        this.operation = Objects.requireNonNull(operation, "operation can't be null");
+        this.resourcePattern = Objects.requireNonNull(resourcePattern, "resourcePattern can't be null");
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/server/authorizer/Action.java:44]: Shall we mention this in the javadoc too?

**Comment by chia7712** [comment]
@ijuma thanks for your suggestion. I have addressed all comments. PTAL

**Review by ijuma — APPROVED** [review]
LGTM, thanks.

**Pull request merged**